packages/salesforcedx-sobjects-faux-generator/src/describe/sObjectDescribe.ts

```diff
@@ -205,12 +252,70 @@ export class SObjectDescribe {
     }).execute();
 
     const cmdOutput = new CommandOutput();
-    const result = await cmdOutput.getCmdResult(execution);
+    let result: string;
+    try {
+      result = await cmdOutput.getCmdResult(execution);
+    } catch (e) {
+      return Promise.reject(e);
+    }
     try {
       const sobjects = JSON.parse(result).result as string[];
       return Promise.resolve(sobjects);
     } catch (e) {
       return Promise.reject(result);
     }
   }
+
+  public async describeSObjectBatch(
+    projectPath: string,
+    types: string[],
+    lastProcessed: number,
+    username?: string
+  ): Promise<SObject[]> {
+    const batchSize = 25;
+
+    await this.setupConnection(projectPath);
+
+    const batchRequest: BatchRequest = { batchRequests: [] };
+
+    for (
+      let i = lastProcessed + 1;
+      i <= lastProcessed + batchSize && i < types.length;
+      i++
+    ) {
+      const urlElements = [this.sobjectsPart, types[i], 'describe'];
+      const requestUrl = urlElements.join('/');
+
+      batchRequest.batchRequests.push({ method: 'GET', url: requestUrl });
+    }
+    const batchUrlElements = [
+      this.instanceUrl,
+      this.servicesPath,
+      this.batchPart
+    ];
+    const batchRequestUrl = batchUrlElements.join('/');
+    const options: XHROptions = {
+      type: 'POST',
+      url: batchRequestUrl,
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+        Authorization: `OAuth ${this.accessToken}`
+      },
+      data: JSON.stringify(batchRequest)
+    };
+
+    try {
+      const response: XHRResponse = await xhr(options);
+      const batchResponse = JSON.parse(response.responseText) as BatchResponse;
```

Comment:
We had talked about storing the raw JSON for validation or future uses. Should we still do that?

Reply:
yes, we should - I just didn't want to try to jam it in - hopefully get this PR in and done :-)...     But yes, we will need it for sure when we do the XML parse step with custom fields...